velox/functions/lib/DateTimeFormatter.cpp

```diff
@@ -295,18 +294,13 @@ bool specAllowsPlusSign(DateTimeFormatSpecifier s, bool specifierNext) {
   }
 }
 
-void parseFail(
-    const std::string_view& input,
-    const char* cur,
-    const char* end,
-    const bool failOnError) {
-  if (failOnError) {
-    VELOX_DCHECK_LE(cur, end);
-    VELOX_USER_FAIL(
-        "Invalid date format: '{}' is malformed at '{}'",
-        input,
-        std::string_view(cur, end - cur));
-  }
+Expected<DateTimeResult>
+parseFail(const std::string_view& input, const char* cur, const char* end) {
+  VELOX_DCHECK_LE(cur, end);
+  return folly::makeUnexpected(Status::UserError(fmt::format(
```

Comment:
No need to use fmt::format. You can just

```
Status::UserError(
      "Invalid date format: '{}' is malformed at '{}'",
      input,
      std::string_view(cur, end - cur));
```

It would be nice to also add a check for threadSkipErrorDetails: https://velox-lib.io/blog/optimize-try-more#do-not-produce-or-store-error-messages-under-try